### ISSUE TITLE
Winget: Cannot Install Zipped Applications

### ISSUE THREAD

**Issue by soulspark666** [microsoft/winget-cli#3416]
### Brief description of your issue

```
winget install Platformtools
```

Returns me with
```
Found PlatformTools [Google.PlatformTools] Version 34.0.3
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Extracting archive...
Failed to extract the contents of the archive
```

### Steps to reproduce

```
winget install Platformtools
```
or
```
winget install MinGit
```

### Expected behavior

This should install without any issues.

### Actual behavior

```
Found PlatformTools [Google.PlatformTools] Version 34.0.3
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Extracting archive...
```
```
Found MinGit [Git.MinGit] Version 2.41.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Extracting archive...
Failed to extract the contents of the archive
```

### Environment

```shell
Windows Package Manager (Preview) v1.6.1573-preview
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1848
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.21.1573.0
```

**Comment by Trenly** [comment]
Can you please attach the log file from this command:
```
winget install Git.MinGit --verbose
```

**Comment by soulspark666** [comment]
Sure here it is.
```
 ~#@❯ winget install Git.MinGit --verbose
Found MinGit [Git.MinGit] Version 2.41.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Extracting archive...
Failed to extract the contents of the archive`
```

**Comment by soulspark666** [comment]
Also attaching the winget logs from
```
winget --logs
```
[WinGet-2023-07-10-11-14-47.140.log](https://github.com/microsoft/winget-cli/files/11998174/WinGet-2023-07-10-11-14-47.140.log)

**Comment by Trenly** [comment]
Thanks!

Seems like the actual error is this -
> D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Archive.cpp(30)\WindowsPackageManager.dll!00007FFFE44A4E52: (caller: 00007FFFE440E925) ReturnHr(1) tid(918) 800700C1 %1 is not a valid Win32 application.

Have you tried re-installing winget to see if the error goes away?

**Comment by soulspark666** [comment]
Yes, I reinstalled it and also Upgraded to `v1.6.1573-preview` in hopes this might be an issue with a specific version

**Comment by Kahana82** [comment]
I just managed to get around .zip packages not installing on my system by editing the UniGetUI settings file, to force the use of the **TAR** executable instead of the one built-in to Windows (**shellApi**), so that archives can unpack successfully during the install process.
My assumption is that disabling the whole built-in ZipFolders thing for Windows Explorer, breaks the use of the default extraction method (shellApi) WinGetUI wants to use.

Edit WinGetUI settings file - from the command prompt:
`winget settings`

then add:
```
  "installBehavior": {
    "archiveExtractionMethod": "tar"
  },
```

There are other interesting settings that can been changed, see documentation [here](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md).


Suggestion to resolve the issue for unaware users:
Add an option to the "Retry" button (after the failed installation attemt) to use TAR instead of shellAPI.

**Comment by Trenly** [comment]
@Kahana82 - Thanks for the feedback. I've cross-posted your comment into the [UniGet UI Repository](https://github.com/marticliment/UniGetUI/issues/3529), as UniGetUI is a completely separate tool from WinGet and WinGet has no UI except for the command line interface.

**Comment by marticliment** [comment]
NOTE: I am the UniGetUI developer.

I am sorry, but UniGetUI does no such thing. In fact, the user is editing WinGet settings, not UniGetUI (they are using the `winget settings` command).

> My assumption is that disabling the whole built-in ZipFolders thing for Windows Explorer, breaks the use of the default extraction method (shellApi) WinGetUI wants to use.

The issue here seems to be the user disabling ZIP support on their system? So the issue is ultimately WinGet not detecting that the zip support is missing, and not fallbacking to the tar extractor.

Either way, UniGetUI does not interfiere on the install/update/uninstall process, but rather calls WinGet with the appropiate command-line parameters.

**Comment by Kahana82** [comment]
> I am sorry, but UniGetUI does no such thing. In fact, the user is editing WinGet settings, not UniGetUI (they are using the `winget settings` command).

You're absolutely right, my appologies for conflating WinGet and UniGetUI.

The issue was definately between WinGet and the built-in ZIP support not being loaded in Windows.
